litmus-portal/frontend/src/views/AnalyticsDashboard/MonitoringDashboardPage/PanelContent.tsx

```diff
@@ -100,66 +118,97 @@ const PanelContent: React.FC<PanelResponse> = ({
-              date: parseInt(dataPoint.timestamp ?? '0', 10) * 1000,
-              value: parseInt(dataPoint.value ?? '0', 10),
-            })),
-            baseColor: areaGraph[index % areaGraph.length],
-          }));
+  const updateGraphData = () => {
+    let seriesData: Array<GraphMetric> = [];
+    let eventData: Array<GraphMetric> = [];
+    if (prometheusData) {
+      prometheusData.GetPromQuery.forEach((queryResponse) => {
+        if (prometheusQueryData.chaosInput.includes(queryResponse.queryid)) {
+          if (queryResponse.legends && queryResponse.legends[0]) {
+            eventData.push(
+              ...queryResponse.legends.map((elem, index) => ({
+                metricName: elem[0] ?? 'chaos',
+                data: queryResponse.tsvs[index].map((dataPoint) => ({
+                  date: parseInt(dataPoint.timestamp ?? '0', 10) * 1000,
+                  value: parseInt(dataPoint.value ?? '0', 10),
+                })),
+                baseColor: areaGraph[index % areaGraph.length],
+              }))
+            );
+          }
+        } else if (queryResponse.legends && queryResponse.legends[0]) {
+          seriesData.push(
+            ...queryResponse.legends.map((elem, index) => ({
+              metricName: elem[0] ?? 'metric',
+              data: queryResponse.tsvs[index].map((dataPoint) => ({
+                date: parseInt(dataPoint.timestamp ?? '0', 10) * 1000,
+                value: parseFloat(dataPoint.value ?? '0.0'),
+              })),
+              baseColor: lineGraph[index % lineGraph.length],
+            }))
+          );
         }
-      }
-    });
-  }
+      });
+      setGraphData({
+        seriesData,
+        eventData,
+      });
+      seriesData = [];
+      eventData = [];
+    }
+  };
 
   useEffect(() => {
-    if (firstLoad === true && updateQueries === false) {
-      generatePrometheusQueryData();
-      setFirstLoad(false);
-      setUpdateQueries(true);
+    if (
+      synchronizer.firstLoad === true &&
+      synchronizer.updateQueries === false
+    ) {
+      if (prom_queries.length) {
+        generatePrometheusQueryData();
+        setSynchronizer({
+          updateQueries: true,
+          firstLoad: false,
+          fetch: true,
+        });
+      }
     }
-    if (updateQueries === true && firstLoad === false) {
+    if (
+      synchronizer.updateQueries === true &&
+      synchronizer.firstLoad === false
+    ) {
       setTimeout(() => {
-        generatePrometheusQueryData();
+        if (prom_queries.length) {
+          generatePrometheusQueryData();
+          setSynchronizer({
+            ...synchronizer,
+            fetch: true,
+          });
+        }
       }, selectedDashboard.refreshRate);
     }
   }, [prometheusQueryData]);
 
-  if (error) {
-    return (
-      <div className={classes.rootPanel}>
-        <Typography>{panel_name}</Typography>
-        <Loader />
-      </div>
-    );
-  }
```

Comment:
If there is an error with the queries, the user will be shown a loader or empty graphs?

Reply:
Empty graph is fine so that he can debug the data source, if we show a loader he may assume that the error is from portal’s end which is not the case as already the graph panel and dashboards load correctly it’s just the Prometheus fetch did not work, also this is a extra overhead to DOM manipulation which may not be too useful and does not convey the error that happened, later we may add a proper error modal or notification for invalid requests or fetch like in other sections.

Reply:
We may take up this in the next sprint